File: src/debug/entry/python3/problems/138.py
def runUserScript(func, params, paramTypes):
    if (len(params) != len(paramTypes)):
        onParameterError()

    newParams = []
    for i, val in enumerate(params):
        newParams.append(parseParameter(i, paramTypes[i], val))
    func(*newParams)


class Node:
    def __init__(self, val, next, random):
        self.val = val
        self.next = next
        self.random = random


def parseNode(param, nodeMap):
    arr = []
    for i, val in enumerate(param):
        arr.append(Node(val[0], None, None))

    for i, val in enumerate(param):
        if i is not len(param) - 1:
            arr[i].next = arr[i + 1]

        if val[1] is not None:
            arr[i].random = arr[val[1]]

    return arr[0]


def parseSpecialParameter(index, paramType, param):
    if paramType == "Node":
        return parseNode(param, None)
    return None

File: src/debug/entry/javascript/problems/138.js
function runUserScript(userFunm, params, paramTypes) {
    if (params.length !== paramTypes.length) {
        onParameterError();
    }

    const parsedParams = params.map((param, index) => {
        const type = paramTypes[index];
        return parseParameter(index, type, param);
    });
    userFunm.apply(null, parsedParams);
}

function Node(val, next, random) {
    this.val = val;
    this.next = next;
    this.random = random;
}

/**
 * @param {number[][]} param
 */
function parseNode(nums) {
    const arr = [];
    nums.map((n, i) => {
        arr.push(new Node(n[0], null, null));
    });
    nums.map((n, i) => {
        if (i !== nums.length - 1) {
            arr[i].next = arr[i + 1];
        }
        if (n[1] !== null) {
            arr[i].random = arr[n[1]];
        }
    });
    return arr[0];
}

function parseSpecialParameter(index, type, param) {
    switch (type) {
        case "Node":
            return parseNode(param);
    }
    return null;
}
